From 7d18349e6bf5dcdad7cef07534bd285c44400e35 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:05:20 -0400
Subject: [usb] ch341: clean up interrupt handler

Message-id: <1410887463-20674-249-git-send-email-dzickus@redhat.com>
Patchwork-id: 95309
O-Subject: [RHEL7 PATCH 248/591] USB: ch341: clean up interrupt handler
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 271ec2d2d7bd4ab528a7a94701503df06ca8fd52
Author: Johan Hovold <jhovold@gmail.com>
Date:   Thu Jan 2 22:49:33 2014 +0100

    USB: ch341: clean up interrupt handler

    Clean up interrupt completion handler somewhat.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/ch341.c b/drivers/usb/serial/ch341.c
index ac21618..65a81c6 100644
--- a/drivers/usb/serial/ch341.c
+++ b/drivers/usb/serial/ch341.c
@@ -484,9 +484,9 @@ static void ch341_update_line_status(struct usb_serial_port *port,
 
 static void ch341_read_int_callback(struct urb *urb)
 {
- struct usb_serial_port *port = (struct usb_serial_port *) urb->context;
+ struct usb_serial_port *port = urb->context;
  unsigned char *data = urb->transfer_buffer;
- unsigned int actual_length = urb->actual_length;
+ unsigned int len = urb->actual_length;
  int status;
 
  switch (urb->status) {
@@ -497,24 +497,23 @@ static void ch341_read_int_callback(struct urb *urb)
  case -ENOENT:
  case -ESHUTDOWN:
   /* this urb is terminated, clean up */
-  dev_dbg(&urb->dev->dev, "%s - urb shutting down with status: %d\n",
+  dev_dbg(&urb->dev->dev, "%s - urb shutting down: %d\n",
    __func__, urb->status);
   return;
  default:
-  dev_dbg(&urb->dev->dev, "%s - nonzero urb status received: %d\n",
+  dev_dbg(&urb->dev->dev, "%s - nonzero urb status: %d\n",
    __func__, urb->status);
   goto exit;
  }
 
- usb_serial_debug_data(&port->dev, __func__,
-         urb->actual_length, urb->transfer_buffer);
- ch341_update_line_status(port, data, actual_length);
+ usb_serial_debug_data(&port->dev, __func__, len, data);
+ ch341_update_line_status(port, data, len);
 exit:
  status = usb_submit_urb(urb, GFP_ATOMIC);
- if (status)
-  dev_err(&urb->dev->dev,
-   "%s - usb_submit_urb failed with result %d\n",
+ if (status) {
+  dev_err(&urb->dev->dev, "%s - usb_submit_urb failed: %d\n",
    __func__, status);
+ }
 }
 
 static int ch341_tiocmget(struct tty_struct *tty)
-- 
1.7.1